crawler-cli/src/main/java/org/gbif/crawler/pipelines/dwca/DwCAToAvroCommandVerification.java

```diff
@@ -0,0 +1,153 @@
+package org.gbif.crawler.pipelines.dwca;
+
+import org.gbif.common.messaging.api.messages.DwcaValidationFinishedMessage;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.Objects;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.eclipse.jdt.internal.core.Assert;
+
+/**
+ * Verifies the configuration parameters like existence of input and output path, and enables with calculated configuration parameters helping DwCAToAvro Conversion
+ */
+public class DwCAToAvroCommandVerification {
+
+  private DwCAToAvroConfiguration configuration;
+  private DwcaValidationFinishedMessage receivedMessage;
+  private DwCA2AvroConfigurationParameter resourceConfigurationParameter;
+
+  public static DwCAToAvroCommandVerification of(DwCAToAvroConfiguration configuration) {
+    return new DwCAToAvroCommandVerification(configuration);
+  }
+
+  private DwCAToAvroCommandVerification(DwCAToAvroConfiguration configuration) {
+    this.configuration = configuration;
+  }
+
+  public DwCAToAvroCommandVerification with(DwcaValidationFinishedMessage receivedMessage) {
+    this.receivedMessage = receivedMessage;
+    return this;
+  }
+
+  /**
+   * verifies parameters and return calculated configuration parameters
+   *
+   * @return Calculated Configuration Parameters
+   */
+  public DwCA2AvroConfigurationParameter verifyParametersAndGetResourceConfigurations() {
+    /*
+      verifies the input configuration and messages are not null
+     */
+    Objects.requireNonNull(configuration, "Configuration cannot be null");
+    Objects.requireNonNull(receivedMessage, "Received message cannot be null");
+    /*
+      checks existence of archiveRepository provided in the yaml config
+     */
+    String inputBasePath = configuration.archiveRepository.endsWith(File.separator)
+      ? configuration.archiveRepository
+      : configuration.archiveRepository.concat(File.separator);
+    Assert.isLegal(Files.exists(new File(inputBasePath).toPath()),
```

Comment:
Assert.isLegal(Files.exists(Paths.get(inputBasePath)))
or 
    Assert.isLegal(new File(inputBasePath).exists(),"Illegal Argument archiveRepository is not a valid path");

?